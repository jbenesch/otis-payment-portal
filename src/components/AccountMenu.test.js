import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import ConnectedAccountMenu, { AccountMenu } from './AccountMenu'

const mockStore = configureStore([])
const store = mockStore({
  user: {
    isLoading: false,
    name: {
      title: 'mr',
      first: 'romain',
      last: 'hoogmoed'
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/83.jpg'
    }
  }
})

it('should render the account menu', () => {
  const wrapper = shallow(
    <AccountMenu
      firstName="Joe"
      avatar="https://randomuser.me/api/portraits/men/51.jpg"
    />
  )
  expect(wrapper).toMatchSnapshot()
})

it("renders the user's first name", () => {
  const wrapper = shallow(
    <AccountMenu
      firstName="Joe"
      avatar="https://randomuser.me/api/portraits/men/51.jpg"
    />
  )
  expect(wrapper.find('b').text()).toEqual('Joe')
})

it('matches a previous snapshot', () => {
  const component = renderer.create(
    <Provider store={store}>
      <ConnectedAccountMenu />
    </Provider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
