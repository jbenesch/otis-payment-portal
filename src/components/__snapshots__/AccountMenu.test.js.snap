// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a previous snapshot 1`] = `
<span
  className="pt-popover-wrapper"
>
  <div
    className="pt-popover-target"
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH gWDJpY"
      disabled={undefined}
      tabIndex={undefined}
    >
      <img
        className="sc-bdVaJa hrdBvl"
        size="large"
        src="https://randomuser.me/api/portraits/men/83.jpg"
        style={
          Object {
            "marginRight": 15,
          }
        }
      />
      <span>
        Hi 
        <b>
          Romain
        </b>
         
        <svg
          className="pt-icon"
          data-icon="chevron-down"
          height={20}
          style={Object {}}
          viewBox="0 0 20 20"
          width={20}
        >
          <title>
            chevron-down
          </title>
          <path
            d="M16 6c-.28 0-.53.11-.71.29L10 11.59l-5.29-5.3a1.003 1.003 0 0 0-1.42 1.42l6 6c.18.18.43.29.71.29s.53-.11.71-.29l6-6A1.003 1.003 0 0 0 16 6z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </div>
</span>
`;

exports[`should render the account menu 1`] = `
<Blueprint2.Popover
  defaultIsOpen={false}
  disabled={false}
  hasBackdrop={false}
  hoverCloseDelay={300}
  hoverOpenDelay={150}
  inheritDarkTheme={true}
  interactionKind="click"
  minimal={false}
  modifiers={Object {}}
  openOnTargetFocus={true}
  position="auto"
  rootElementTag="span"
  targetElementTag="div"
  transitionDuration={300}
  usePortal={true}
>
  <styled.div>
    <styled.img
      size="large"
      src="https://randomuser.me/api/portraits/men/51.jpg"
      style={
        Object {
          "marginRight": 15,
        }
      }
    />
    <span>
      Hi 
      <b>
        Joe
      </b>
       
      <Blueprint2.Icon
        icon="chevron-down"
        iconSize={20}
      />
    </span>
  </styled.div>
  <Blueprint2.Menu>
    <Blueprint2.MenuItem
      disabled={false}
      icon="id-number"
      multiline={false}
      onClick={[Function]}
      popoverProps={Object {}}
      shouldDismissPopover={true}
      text="Edit My Profile"
    />
    <Blueprint2.MenuDivider />
    <Blueprint2.MenuItem
      disabled={false}
      icon="log-out"
      multiline={false}
      onClick={[Function]}
      popoverProps={Object {}}
      shouldDismissPopover={true}
      text="Logout"
    />
  </Blueprint2.Menu>
</Blueprint2.Popover>
`;
